OpenNefia.Content/CharaMake/CharaMakeRaceClassLayer.cs

```diff
@@ -0,0 +1,379 @@
+
+            DetailText = new UiWrapText(450, UiFonts.ListTitleScreenText);
+            DetailContainer.AddElement(DetailText);
+            DetailContainer.AddLayout(LayoutType.YMin, 110);
+
+            DetailContainer.AddLayout(LayoutType.XOffset, -3);
+            DetailContainer.AddElement(AttributeTopic, LayoutType.XOffset, -7);
+
+            DetailContainer.AddLayout(LayoutType.Spacer, 18);
+
+            AttributeContainer = new UiGridContainer(GridType.Horizontal, 3, xCentered: false, xSpace: 50, ySpace: 3);
+            DetailContainer.AddElement(AttributeContainer);
+
+            DetailContainer.AddLayout(LayoutType.Spacer, 5);
+
+            DetailContainer.AddElement(SkillTopic, LayoutType.XOffset, -7);
+            SkillContainer = new UiVerticalContainer();
+            DetailContainer.AddElement(SkillContainer);
+
+            TrainedSkillContainer = new UiVerticalContainer();
+            DetailContainer.AddLayout(LayoutType.Spacer, 18);
+            DetailContainer.AddElement(TrainedSkillContainer);
+
+            OnKeyBindDown += CharaMakeRaceClassLayer_OnKeyBindDown;
+
+            List = new UiList<RaceClass>();
+            List.EventOnActivate += (_, args) =>
+            {
+                Select(args.SelectedCell.Data);
+            };
+            List.EventOnSelect += (_, args) =>
+            {
+                SelectData(args.SelectedCell.Data);
+            };
+
+            AddChild(List);
+            AddChild(Window);
+        }
+
+        public override void Initialize(CharaMakeData args)
+        {
+            base.Initialize(args);
+            AllData = GetData().Select(x => new RaceClassCell(x)).ToArray();
+
+            PageModel.OnPageChanged += Window_OnPageChanged;
+            PageModel.Initialize(AllData);
+            PageModel.SetWindow(Window);
+
+            SelectData(AllData.First().Data);
+        }
+
+        public abstract IEnumerable<RaceClass> GetData();
+
+        private void Window_OnPageChanged()
+        {
+            List.Clear();
+            List.AddRange(PageModel.GetCurrentElements());
+            List.Select(List.SelectedIndex);
+        }
+
+        public override void OnFocused()
+        {
+            base.OnFocused();
+            List.GrabFocus();
+        }
+
+        private void CharaMakeRaceClassLayer_OnKeyBindDown(GUIBoundKeyEventArgs args)
+        {
+            if (args.Function == EngineKeyFunctions.UINextPage)
+            {
+                PageModel.PageForward();
+            }
+            if (args.Function == EngineKeyFunctions.UIPreviousPage)
+            {
+                PageModel.PageBackward();
+            }
+        }
+
+        protected abstract void Select(RaceClass item);
+
+        private void SelectData(RaceClass data)
+        {
+            SetAttributes(data.GetSkills());
+            SetTrainedSkills(data.GetSkills());
+            DetailContainer.Resolve();
+            DetailText.Text = data.GetString("Description");
```

Comment:
This should be a blank string if the description doesn't exist (it shows `<Missing key: ...>` otherwise)

Reply:
done in dd52918